### PR TITLE
fix(DirectiveResolver): throw error on invalid query selector

### PR DESCRIPTION
**Pull request by mgechev**
Fix for https://github.com/angular/angular/issues/4489

### CODE REVIEW

**Comment by ericmartinezr** [comment]
Nice, thank you :+1:

File: modules/angular2/src/core/linker/directive_resolver.ts [hunk 1]
```diff
@@ -97,16 +99,29 @@ export class DirectiveResolver {
...
-    return this._merge(dm, inputs, outputs, host, queries);
+    return this._merge(dm, inputs, outputs, host, queries, type);
+  }
+
+  private _mergeQueries(dm: DirectiveMetadata, queries: {[key: string]: any}, type: Type):
+      {[key: string]: any} {
+    var mergedQueries =
+        isPresent(dm.queries) ? StringMapWrapper.merge(dm.queries, queries) : queries;
+    StringMapWrapper.forEach(mergedQueries, (query: any, propName: string) => {
+      if (!isValidDirective(query.selector)) {
```
Comment (tbosch) [inline modules/angular2/src/core/linker/directive_resolver.ts]: This is wrong. query.selector can contain a string, in which case it selects elements on which a variable with a name was placed.

**Comment by tbosch** [comment]
@mhevery I think this is wrong (see my comment above)

**Comment by mhevery** [comment]
@mgechev Can you look into https://github.com/angular/angular/pull/4548#discussion_r43458440 and provide a fix.

**Comment by mgechev** [comment]
Sure! I'll provide a fix in a few hours.

File: modules/angular2/src/core/linker/directive_resolver.ts (hunk 1 above)
Comment (mgechev) [inline modules/angular2/src/core/linker/directive_resolver.ts]: @tbosch basically I need to include one extra check `if (!isValidDirective(query.selector) && !isString(query.selector)) {...}`. Is my understanding correct?

**Comment by mgechev** [comment]
@mhevery @tbosch I added an extra check to verify that the selector is not a valid directive, neither a string. Is it good now or there's still something missing?

**Comment by mgechev** [comment]
I'll work on fixing the conflicts now.

**Comment by mgechev** [comment]
I accidentally removed my fork so wasn't able to resolve the merge conflicts in this PR. I'm closing this issue in favor of https://github.com/angular/angular/pull/5049.

**Pull request closed without merging**